awd/article.py

```diff
@@ -32,11 +39,26 @@ def __init__(
             metadata_url: The URL for retrieving metadata records.
             content_url: The URL for retrieving article content.
             doi_table: The DOI table as a PynamoDB object.
+            s3_client: A configured S3 client.
+            bucket: The S3 bucket for uploading metadata and article content.
+            sqs_client: A configured SQS client.
+            sqs_base_url: The SQS base URL to use. Enables easier unit testing.
+            sqs_input_queue: The SQS input queue to use.
+            sqs_output_queue: The SQS output queue to use.
+            collection_handle: The handle of the DSpace collection to which items
+            will be uploaded.
         """
         self.doi: str = doi
         self.metadata_url: str = metadata_url
         self.content_url: str = content_url
         self.doi_table: DoiProcessAttempt = doi_table
+        self.s3_client: S3 = s3_client
+        self.bucket: str = bucket
+        self.sqs_client: SQS = sqs_client
+        self.sqs_base_url: str = sqs_base_url
+        self.sqs_input_queue: str = sqs_input_queue
+        self.sqs_output_queue: str = sqs_output_queue
```

Comment:
It's minor, but this raised some flags for me.

I think it's okay in this context, but maybe be wary of affixing clients like this to a class during initialization.  This can make initializing the class a headache sometimes, particularly if you don't need to use those clients.

An alternative approach could be to send along any data or configs those clients might need -- e.g. noticed the SQS client needs the AWS region -- but then init new instances of those clients as needed in the `Article` methods.  

 Again, don't think it needs changing, just noting.

Reply:
This is a good point but I'm assuming that instantiating a new client for each `Article` would slow things down a bit but this is great to consider with future projects!